### ISSUE TITLE
MCMC Optimizer for GaussianSplatting

### ISSUE THREAD

**Issue by fwilliams** [openvdb/fvdb-reality-capture#24]
Implement the MCMC optimization algorithm [https://ubc-vision.github.io/3dgs-mcmc/](https://ubc-vision.github.io/3dgs-mcmc/).

This algorithm allows us to set a target number of gaussians to train to (rather than a fixed iteration count). This lets us have saner defaults based on the total number of pixels in the training set.

It also lets us implement training with the Unscented transform since we don't need to retain the means2d gradients.

**Comment by harrism** [comment]
Sorry, didn't see this before I created #180. There's a bit more info in that including sub issues, so I'm going to close this.
